src/main/java/com/ericsson/eiffel/remrem/publish/service/MessageServiceRMQImpl.java

```diff
@@ -168,4 +208,130 @@ private void createFailureResult(List<PublishResultItem> events) {
                 PropertiesConfig.INVALID_EVENT_CONTENT);
         events.add(event);
     }
+    
+    
+    /**
+     * This method handles multiple events status codes
+     * @param routingKeyMap contains the eventId and routing key of that event
+     * @param msgService Messaging service
+     * @param json
+     * @param userDomainSuffix
+     */
+    public void handleStatusCodesForMultipleEvents(Map<String,String> routingKeyMap,MsgService msgService,
```

Comment:
I don't understand this code. The method naming is wrong, this isn't a handleStatus code it is a send method but that is a minor issue.
Publish should send the messages in the order of the JsonArray they come in.
If a sending of a message fails then the remaining messages in the array should not be sent. Suggested sub status code for remaining messages is 503 but perhaps something better exist? The reason for stopping is that Publish CAN NOT know if there are any dependencies from later events to the failed event. Publish DO NOT know of any specifics of a messaging protocol so it can not traverse inputEventIds (which would only cover a part of one known messaging protocol) or other. The task of traversing events could be put on the messaging protocol library which has this knowledge but lets keep it simple.